esp-hal/src/spi/master.rs

```diff
@@ -1374,6 +1425,41 @@ mod dma {
                 return Err((Error::MaxDmaTransferSizeExceeded, self, buffer));
             }
 
+            #[cfg(all(esp32, spi_address_workaround))]
+            {
+                // On the ESP32, if we don't have data, the address is always sent
+                // on a single line, regardless of its data mode.
+                if bytes_to_write == 0 && address.mode() != SpiDataMode::Single {
+                    let bytes_to_write = address.width().div_ceil(8);
+                    let addr_bytes = address.value().to_le_bytes();
```

Comment:
Don't be hasty, I need to verify if I got the endianness right here.

Reply:
iirc the bit order of the data phase can changed so you'll probably have to read that register bit to decide the endianness here.

Reply:
SPI_WR_BIT_ORDER affects all (CMD, ADDR and DATA) phases so I think I'm fine here.

Reply:
That being said, I wanted to look at this with a logic analyzer because this workaround exists due to the hardware not doing what it's supposed to.

Reply:
Good thing I checked. The bit works as expected, but the address is encoded as big-endian so that had to be aligned.

Demo that we now generate the same waveform with or without data. 2 lines only, because my ESP32 devkit doesn't have enough pins 😅

![image](https://github.com/user-attachments/assets/13ddbf9d-a106-475f-9ab3-22f1ac4b2b1a)

Reply:
Nice!

I just remembered, will this work as the user expects if they specify a dummy phase?

Reply:
No, the dummy phase gets moved to before the address which is wrong. But that'll need to be addressed later as there are a few different approaches to deal with that (error, ignore, emulate). This current PR is done the way it is.